botocore/loaders.py

```diff
@@ -461,16 +564,17 @@ def _potential_locations(self, name=None, must_exist=False, is_dir=False):
         # Will give an iterator over the full path of potential locations
         # according to the search path.
         for path in self.search_paths:
-            if os.path.isdir(path):
+            if path.is_dir():
                 full_path = path
                 if name is not None:
-                    full_path = os.path.join(path, name)
+                    name_parts = name.split(os.sep)
+                    full_path = path.joinpath(*name_parts)
```

Comment:
this is required to handle windows paths i.e `\foo\bar\test.` Even on windows, `zipfile` uses posix path separators `/`